Assets/Scripts/CharacterController.cs

```diff
@@ -29,7 +29,7 @@ void Start()
     {
         rigidbody = GetComponent<Rigidbody2D>();
         playerSpriteRenderer = GetComponent<SpriteRenderer>();
-        //playerSpawn = GameObject.Find("Player Spawn").GetComponent<GameObject>();
+        playerSpawn = GameObject.Find("Player Spawn").GetComponent<Transform>();
```

Comment:
if you have multiple spawns/checkpoints in the same map, consider changing this to just a regular getcomponent and using triggers to change the current spawn point of the player